utils/extract_strings.pl

```diff
@@ -137,7 +137,7 @@ sub extract_target_strings {
                 $str = "";
                 $state = 1;
                 next;
-            } elsif(/^Contents/ && ! /\.ro?data/) {
+            } elsif(/^Contents of/) {
                 $str = "";
```

Comment:
why is this one a problem, but `if(/section \.ro?data/ || /section __cstring/) {` is ok?

Reply:
the following line is not matched with this pattern so it is trying to get parsed:
Contents of section .rel.rodata.devo_opts:

Generates the following warnings:
Illegal hexadecimal digit 'o' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'n' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'n' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 't' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'o' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 's' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 't' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'i' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'n' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'r' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'l' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'r' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 't' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit '.' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'v' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'o' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 'o' ignored at ../utils/extract_strings.pl line 159.
Illegal hexadecimal digit 't' ignored at ../utils/extract_strings.pl line 159.

Reply:
On linux, same content is headed by:
Contents of section .rodata.devo_opts:
 0000 00000000 0a000000 0e000000 13000000  ................
 0010 00000000 00000000

Reply:
I think you want to match his in the 1st match (state==1) rather than the 2nd match (state == 0).

Something like:
```
if(/section (\.rel)?\.ro?data/ || /section __cstring/) {
```